### PR TITLE
Updating RDatasetSpec docs with Cpp MWE

### PR DESCRIPTION
**Pull request by edfink234**
# This Pull request:

## Checklist:

- [ ] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #

### CODE REVIEW

File: tree/dataframe/src/RDataFrame.cxx [hunk 1]
```diff
@@ -1466,6 +1466,30 @@ RDataFrame::RDataFrame(std::unique_ptr<ROOT::RDF::RDataSource> ds, const ColumnN
 /// spec.AddFriend([("tree1", "a.root"), ("tree2", "b.root")], "alias")
 /// df = ROOT.RDataFrame(spec)
 /// ~~~
+///
+/// ~~~{.cpp}
+/////First, create a dummy root file with two trees,
```
Comment (vepadulano) [inline tree/dataframe/src/RDataFrame.cxx:1471]: The example code should have a whitespace between the three forward slashes that are needed for doxygen and the rest of the body, as found in lines above

File: tree/dataframe/src/RDataFrame.cxx [hunk 2]
```diff
@@ -1466,6 +1466,30 @@ RDataFrame::RDataFrame(std::unique_ptr<ROOT::RDF::RDataSource> ds, const ColumnN
...
+/////(w/ cols z)
+///ROOT::RDataFrame df(10); //Empty dataframe of ten rows
+///auto columns = df.Define("x", []{return 1;}, {}).Define("y", "2").Define("z", "3"); //define 3 columns x, y, z
+///columns.Display({"x","y"}, 10)->Print(); //print out first 10 rows
+/////snap columns x and y to outputTree1 and column z
+/////to outputTree2 in outputFile.root
+///columns.Snapshot("outputTree1", "outputFile.root", {"x", "y"});
+///ROOT::RDF::RSnapshotOptions opts;
+///opts.fMode = "update";
+///columns.Snapshot("outputTree2", "outputFile.root", {"z"}, opts);
```
Comment (vepadulano) [inline tree/dataframe/src/RDataFrame.cxx:1482]: I am curious as to why you think that adding this portion of the example is relevant for this particular constructor. This looks more like the beginning of a short tutorial, doesn't it?

File: tree/dataframe/src/RDataFrame.cxx [hunk 3]
```diff
@@ -1466,6 +1466,30 @@ RDataFrame::RDataFrame(std::unique_ptr<ROOT::RDF::RDataSource> ds, const ColumnN
...
+/////Now create the RDataFrame
+///ROOT::RDF::Experimental::RDatasetSpec x;
+///std::vector<std::string> files = {"outputFile.root"};
+/////create RSample object with name myPhysicsDf and tree outputTree1
+///ROOT::RDF::Experimental::RSample r("myPhysicsDf", "outputTree1", files);
+///x.AddSample(r); //add RDatasetSpec to RDatasetSpec
+///x.WithGlobalFriends("outputTree2", files); //add friend tree outputTree2
+///ROOT::RDataFrame df(x); //create RDataFrame df
+///df.Display({"x","y","z"}, 10)->Print(); //all three columns are printed
```
Comment (vepadulano) [inline tree/dataframe/src/RDataFrame.cxx:1491]: The inner details of the creation of an `RDatasetSpec` are probably better off in the documentation of `RDatasetSpec` itself. I don't think that this interface will anyway be the "suggested" one for 99% of users, that's going to be creating the RDF via `FromSpec`.

**Review by vepadulano — CHANGES_REQUESTED** [review]
Hi @edfink234 ,

Thanks for giving this a shot. I have placed some comments below. I think the idea of placing detailed explanation about how to create an `RDatasetSpec` should be placed in its own documentation. I don't think this constructor should publicize this too much. Maybe a link to the class itself could be enough. See the rest of the comments below as well.

**Review by vepadulano — CHANGES_REQUESTED** [review]
Admittedly the example is not the most common one, having one main file with the main tree and one separate friend file with the friend tree would be more common I believe. Still, it is a good example and written in the correct place, thanks for that!

Before merging, can you squash the 2 commits into one? Afterwards, we can merge this PR.

**Comment by edfink234** [comment]
I think I did it...

**Review by vepadulano — APPROVED** [review]

**Pull request merged**